STYLE.md

```diff
@@ -0,0 +1,453 @@
+* Comment every exported data type.
+
+Function example:
+
+```haskell
+-- | Send a message on a socket. The socket must be in a connected
+-- state. Returns the number of bytes sent. Applications are
+-- responsible for ensuring that all data has been sent.
+send
+  :: Socket      -- ^ Connected socket
+  -> ByteString  -- ^ Data to send
+  -> IO Int      -- ^ Bytes sent
+```
+
+* For functions, the documentation should give enough information to
+apply the function without looking at its definition.
+
+Record example:
+
+```haskell
+-- | Bla bla bla.
+data Person = Person
+  { age :: !Int     -- ^ Age
+  , name :: !String  -- ^ First name
+  }
+```
+
+For fields that require longer comments, format them this way:
+
+```haskell
+data Record = Record
+  { -- | This is a very very very long comment that is split over
+    -- multiple lines.
+    field1 :: !Text
+
+    -- | This is a second very very very long comment that is split
+    -- over multiple lines.
+  , field2 :: !Int
+  }
+```
+
+#### End-of-Line Comments
+
+Separate end-of-line comments from the code with 2 spaces. Align
+comments for data type definitions. Some examples:
+
+```haskell
+data Parser =
+  Parser
+    !Int         -- ^ Current position
+    !ByteString  -- ^ Remaining input
+
+foo :: Int -> Int
+foo n = salt * 32 + 9
+  where
+    salt = 453645243  -- Magic hash salt.
+```
+
+#### Links
+
+Use in-line links economically.  You are encouraged to add links for
+API names. It is not necessary to add links for all API names in a
+Haddock comment. We therefore recommend adding a link to an API name
+if:
+
+* The user might actually want to click on it for more information (in
+  your opinion), and
+
+* Only for the first occurrence of each API name in the comment (don't
+  bother repeating a link)
+
+Top-down guideline
+------------------
+
+### LANGUAGE extensions section
+
+Write each `LANGUAGE` pragma on its own line, sort them alphabetically and
+align by max width among them.
+
+```haskell
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE Rank2Types #-}
+{-# LANGUAGE TemplateHaskell #-}
+```
+
+### Module name
+
+Use singular when naming modules (e.g. use `Data.Map` and
+`Data.ByteString.Internal` instead of `Data.Maps` and
+`Data.ByteString.Internals`). Sometimes it's acceptable to use plural
+(e. g. `Types`, `Instances`).
+
+### Export Lists
+
+Format export lists as follows:
+
+```haskell
+module Data.Set
+  ( Set
+  , empty
+  , singleton
+  , member
+  ) where
+```
+
+Some clarifications:
+
+1. Use 2 spaces indentation for export list.
+2. You can split export list into sections or just write all as single section.
+3. It is strongly adviced to sort each section alpabetically. However,
+   classes, data types and type aliases should be written before
+   functions.
+
+### Imports
+
+Imports should be grouped in the following order:
+
+0. Implicit import of custom prelude (for example [`cardano-prelude`](https://github.com/input-output-hk/cardano-prelude)) if used.
```

Comment:
```suggestion
0. Import of custom prelude (for example [`cardano-prelude`](https://github.com/input-output-hk/cardano-prelude)) if used.
```
Isn't it not implicit, if it is there? e.g.
```
import Cardano.Prelude
```
https://github.com/input-output-hk/cardano-ledger/blob/677b9377d4dc541eb3d7ffac079bdab7d489df1f/cardano-ledger/src/Cardano/Chain/Epoch/File.hs#L13

Reply:
Good catch, I'll fix.